utils/types/deployment/deployment.go

```diff
@@ -33,3 +38,39 @@ func (t Type) Valid() bool {
 	}
 	return false
 }
+
+func GetConnectionToken() (string, bool, error) {
+	deploymentType, err := GetFromEnv()
+	if err != nil {
+		pkgLogger.Errorf("error getting deployment type: %s", err.Error())
+		return "", false, err
+	}
+	var connectionToken string
+	var isMultiWorkspace bool
+	switch deploymentType {
+	case DedicatedType:
+		connectionToken = config.GetWorkspaceToken()
+	case MultiTenantType:
+		isMultiWorkspace = true
+		isNamespaced := config.IsEnvSet("WORKSPACE_NAMESPACE")
+		if isNamespaced {
+			connectionToken, err = config.GetEnvErr("WORKSPACE_NAMESPACE")
+			if err != nil {
+				pkgLogger.Errorf("error getting workspace namespace: %s", err.Error())
+				return "", false, err
+			}
+			if connectionToken == HostedNamespace {
+				// CP Router still has some things hardcoded for hosted
+				// which needs to be supported
+				connectionToken = config.GetEnv("HOSTED_SERVICE_SECRET", "")
+			}
+		} else {
```

Comment:
[optional] This block is very long, you could do a `break` at the end of the above `if` and remove this `else`.